### PR TITLE
ftr(scan): include commit_timestamp in ci

### PR DESCRIPTION
**Pull request by ThatDevShparki**
## Context
In Semgrep Cloud Platform, scans are used to capture contributor information for `semgrep ci` initiated scans. A component of this as been assuming that the commit associated with the scan occurred at relatively the same time as the scan began. This is not always the case when there are scheduled scans on older repos with ancient commits.

## Solution
Include the `commit_timestamp` along with other commit information sent in `scan.meta`

Test Plan:
- Update snapshots to show that we are now including `commit_timestamp` in the `scan.meta` object

PR checklist:

- [X] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [X] Tests included or PR comment includes a reproducible test plan
- [X] Documentation is up-to-date
- [X] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [ ] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by ievans** [comment]
👍 @ThatDevShparki does this require an update to PRIVACY.md as well?

**Comment by ThatDevShparki** [comment]
> 👍 @ThatDevShparki does this require an update to PRIVACY.md as well?

Great callout, just updated it! It seems like we already mention we capture timestamps, but I added to the commit metadata section that we are now explicitly capturing the timestamp along with the commit hash.

File: PRIVACY.md
Comment (chmccreery) [inline PRIVACY.md:58]: This language is confusing - as we then go on to say that it will leave your environment under certain conditions

File: cli/src/semgrep/meta.py
Comment (chmccreery) [inline cli/src/semgrep/meta.py]: So this was already collected and defined and sent to the app as part of each Finding - do you agree? Looking at `RuleMatch.to_app_finding_format()`

File: cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-azure-pipelines-overwrite-autodetected-variables/meta.json
Comment (chmccreery) [inline cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-azure-pipelines-overwrite-autodetected-variables/meta.json]: I always prefer human-readable timestamps, but that's personal preference. Either way, it would be helpful to include explicit timezone information here (if it's always UTC, call that out, if not, label with timezone)

**Review by chmccreery — COMMENTED** [review]
Code looks good to me - might need another review/approval from the privacy side

File: cli/src/semgrep/meta.py
Comment (ThatDevShparki) [reply cli/src/semgrep/meta.py]: Yes, this was sent with each Finding, we just want to include it in the scan meta as well moving forward

File: PRIVACY.md
Comment (ThatDevShparki) [reply PRIVACY.md:58]: It was confusing to me as well. I opted to not update the language outside the scope of these changes but that should maybe a rewording here would be nice. I can ping technical writers to check in on this

**Review by chmccreery — APPROVED** [review]
Approving to unblock, but I do feel that the time-zone comment is important to resolve. I think it probably makes sense to convert it to UTC for consistency with other timestamps that are saved in the app.

File: cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-azure-pipelines-overwrite-autodetected-variables/meta.json
Comment (ThatDevShparki) [reply cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-azure-pipelines-overwrite-autodetected-variables/meta.json]: I've updated the code the match the same format that we use with findings, something similar to `2023-07-24T23:38:18`

**Comment by werrett** [comment]
LGTM. No concerns around collecting timestamps like this.

**Review by werrett — APPROVED** [review]

**Pull request merged**
